contracts/zero-ex/contracts/src/features/ERC721OrdersFeature.sol

```diff
@@ -0,0 +1,1112 @@
+            //         so, we revert in the following step.
+            uint256 sellOrderFees = _payFees(sellOrder, buyOrder.maker, false);
+            
+            // Step 4: The spread must be enough to cover the sell order fees.
+            //         If not, `_payFees` will have taken more tokens from the 
+            //         buyer than they had agreed to in the buy order, in which 
+            //         case we revert here.
+            rrequire(
+                spread >= sellOrderFees,
+                LibERC721OrdersRichErrors.SellOrderFeesExceedSpreadError(
+                    sellOrderFees,
+                    spread
+                )
+            );
+
+            // Step 7: We calculate the profit as:
+            //         profit = buyOrder.erc20TokenAmount - sellOrder.erc20TokenAmount - sellOrderFees
+            //                = spread - sellOrderFees
+            //         I.e. the buyer would've been willing to pay up to `profit`
+            //         more to buy the asset, so instead that amount is sent to
+            //         `msg.sender` as the profit from matching these two orders.
+            profit = spread - sellOrderFees;
+            if (profit > 0) {
+                _transferERC20TokensFrom(
+                    buyOrder.erc20Token,
+                    buyOrder.maker,
+                    msg.sender,
+                    profit
+                );
+            }
+        }
+
+        // TODO: or we could emit 2 ERC721OrderFilled events
+        emit ERC721OrdersMatched(
+            sellOrder.erc20Token,
+            sellOrder.erc721Token,
+            sellOrder.erc20TokenAmount,
+            sellOrder.erc721TokenId,
+            sellOrder.maker,
+            buyOrder.maker,
+            sellOrder.nonce,
+            buyOrder.nonce,
+            msg.sender,
+            profit
+        );
+    }
+    
+    /// @dev Matches pairs of complementary orders that have
+    ///      non-negative spreads. Each order is filled at 
+    ///      their respective price, and the matcher receives
+    ///      a profit denominated in the ERC20 token.
+    /// @param sellOrders Orders selling ERC721 assets.
+    /// @param buyOrders Orders buying ERC721 assets.
+    /// @param sellOrderSignatures Signatures for the sell orders.
+    /// @param buyOrderSignatures Signatures for the buy orders.
+    /// @return profits The amount of profit earned by the caller
+    ///         of this function for each pair of matched orders
+    ///         (denominated in the ERC20 token of the order pair).
+    /// @return successes An array of booleans corresponding to
+    ///         whether each pair of orders was successfully matched.
+    function batchMatchERC721Orders(
+        LibERC721Order.ERC721Order[] memory sellOrders,
+        LibERC721Order.ERC721Order[] memory buyOrders,
+        LibSignature.Signature[] memory sellOrderSignatures,
+        LibSignature.Signature[] memory buyOrderSignatures
```

Comment:
should *this* have a `revertIfIncomplete`?

Reply:
idk, I can't really think of a reason people would ever set `revertIfIncomplete` to true?

Reply:
Yeah, I guess. I added it in my branch but you can take it out.